src/Controller/MobileController.php

```diff
@@ -4,11 +4,37 @@
 namespace App\Controller;
 
 
+use App\Entity\Mobile;
 use FOS\RestBundle\Controller\Annotations as Rest;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
 
 class MobileController extends AbstractController
 {
+    /**
+     * Return mobile detail
+     * @Rest\Get(
+     *     path="/api/mobiles/{id}",
+     *     requirements = {"id"="\d+"}
+     * )
+     * @Rest\View(
+     *     StatusCode = 200,
+     *     serializerGroups={"mobile"}
+     * )
+     * @param Mobile $mobile
+     * @return Mobile
+     */
+    public function mobile(Mobile $mobile)
+    {
+        $users = $mobile->getUsers();
+        foreach ($users as $user) {
```

Comment:
public function mobile(Mobile $mobile)
{
if (!$mobile->getUser()->contains($this->getUser()) {
    throw new AccessDeniedHttpException('This mobile does not belong to you');
}

return $mobile;
}

C'est pas plus propre comme ça ? :D